pkg/forwarder/transaction/transaction.go

```diff
@@ -335,6 +335,14 @@ func (t *HTTPTransaction) internalProcess(ctx context.Context, client *http.Clie
 		tlmTxHTTPErrors.Inc(t.Domain, transactionEndpointName, statusCode)
 	}
 
+	limit := 100
+	if limit < len(body) {
+		// Truncate HTTP response body by first 100 bytes
+		// to tell which of those tools a response comes
+		// and prevent from logging a huge message.
+		body = body[:limit]
+	}
+
```

Comment:
Following on https://github.com/DataDog/datadog-agent/pull/10100#issuecomment-987943539, add logic to truncate a HTTP response body.

https://go.dev/play/p/hyrKIKQtm6x is test code.

Reply:
Looks good. Let's move this to a separate function though, and call it directly on the line with `Errorf` below. And 100 might be a bit too small, maybe 1000?

Reply:
Does it make sense to you? https://github.com/DataDog/datadog-agent/pull/10100/commits/13f56662fe45972a4115f5ba48df9b51baa66dff

I am just wondering if I should move it to a project wide function because it is not only for this time.

Reply:
Not quite what I had in mind. `body` is also returned from the function, so we should not just truncate it early (at least not without checking that all consumers would be fine with it, and documenting the new behavior). For the purposes of this PR, because you just want a better log message, we can limit the size only when logging: `log.Errorf("...: %q", ..., truncateBodyForLog(body))`